From c34dbe64c02ceedb8511d1543ba8b97cede428cf Mon Sep 17 00:00:00 2001
From: Tirumalesh Chalamarla <tchalamarla@cavium.com>
Date: Fri, 3 Apr 2015 13:30:11 -0700
Subject: [PATCH] ARM64/KVM : Add support for 48bit PA in EL2
Patch-Mainline: Not yet, this is to be done by Cavium
References: fate#319484

as the Kernel is free to use 48bit PA, systems with high amount of RAM
or multinode systems can create EL2 page tables at PA greater than 40bits.
This results in unexpected problems with EL2 page tables translations.

This patch  programs TTBR0_EL2 with 48 bit output size.

Signed-off-by: Tirumalesh Chalamarla <tchalamarla@cavium.com>
Signed-off-by: Robert Richter <rrichter@cavium.com>
Signed-off-by: Matthias Brugger <mbrugger@suse.com>
---
 arch/arm64/include/asm/kvm_arm.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/include/asm/kvm_arm.h b/arch/arm64/include/asm/kvm_arm.h
index ac6fafb..a896a5c 100644
--- a/arch/arm64/include/asm/kvm_arm.h
+++ b/arch/arm64/include/asm/kvm_arm.h
@@ -98,6 +98,7 @@
 #define TCR_EL2_TBI	(1 << 20)
 #define TCR_EL2_PS	(7 << 16)
 #define TCR_EL2_PS_40B	(2 << 16)
+#define TCR_EL2_PS_48B	(5 << 16)
 #define TCR_EL2_TG0	(1 << 14)
 #define TCR_EL2_SH0	(3 << 12)
 #define TCR_EL2_ORGN0	(3 << 10)
@@ -106,7 +107,7 @@
 #define TCR_EL2_MASK	(TCR_EL2_TG0 | TCR_EL2_SH0 | \
 			 TCR_EL2_ORGN0 | TCR_EL2_IRGN0 | TCR_EL2_T0SZ)
 
-#define TCR_EL2_FLAGS	(TCR_EL2_PS_40B)
+#define TCR_EL2_FLAGS	(TCR_EL2_PS_48B)
 
 /* VTCR_EL2 Registers bits */
 #define VTCR_EL2_PS_MASK	(7 << 16)
-- 
2.5.0.234.gefc8a62

